### ISSUE TITLE
Sending Http calls to API in container from localhost

### ISSUE THREAD

**Issue by nealhartley** [nealhartley/leashed#3]
Attempting to make any HTTP request to the API in the container results in getting no response from the server. Not an error code or anything. This infers that there is an issue with routing internal to the docker container as any request, correct or not, should result in an error code if it has made it to the Application.

**Comment by nealhartley** [comment]
I believe this is an issue with the applications setttings. The launchsettings set it to listen to "http://localhost:xxxx" and also "https://locahost:xxxx+1" We had the issue when trying to route to the sql server via localhost and found we needed to change it to the applications domain as given by docker-compose. Now working out how to get the dotnet application to listen on its location instead.

**Comment by nealhartley** [comment]
To look at when I get home:
https://www.reddit.com/r/dotnet/comments/8hmjin/exposing_aspnet_core_port_via_docker/

**Comment by nealhartley** [comment]
Docker was setting to point at localhost:5000 this was not correct. We needed to bind to *:5000. Dotnet looks in multiple places when it starts to build up the kestrel https server. The first place it looks is the environement variables. Then it moves down to configuration variables, and then further down. 

To fix this problem I added an environement variable to the docker file that spins up the app. 
The required line was: "ENV ASPNETCORE_URLS http://*:5000"
Now when the app spins up it will respond to anything hitting its container on any ip at port 5000.

I will look into extending this adding variables for https and look into ipv4 and ipv6 bindings.
